reflekt-plugin/src/main/kotlin/io/reflekt/plugin/analysis/processor/uses/FunctionUsesProcessor.kt

```diff
@@ -23,4 +27,17 @@ class FunctionUsesProcessor(override val binding: BindingContext, private val re
     }
 
     override fun shouldRunOn(element: KtElement) = element.isPublicFunction
+
+    private fun SignatureToAnnotations.covers(function: KtNamedFunction): Boolean =
+        (annotations.isEmpty() || function.getAnnotations(binding, annotations).isNotEmpty()) && checkSignature(function)
+
+    private fun SignatureToAnnotations.checkSignature(function: KtNamedFunction): Boolean {
+        val argumentTypes = function.argumentTypesWithReceiver(binding)
+        val returnType = function.returnType(binding) ?: error("No return type???")
```

Comment:
Please, use more polite error :) I think we can add todo because now I am not sure should we throw an exception or not